web/pdf_page_view.js

```diff
@@ -1189,4 +1299,240 @@ class PDFPageView {
+      // .maxCanvasPixels will at least have enough pixels to cover the visible
+      // screen.
+    }
+
+    const overflowWidth = visibleWidth * overflowScale;
+    const overflowHeight = visibleHeight * overflowScale;
+
+    const minX = Math.max(0, visibleArea.minX - overflowWidth);
+    const maxX = Math.min(viewport.width, visibleArea.maxX + overflowWidth);
+    const minY = Math.max(0, visibleArea.minY - overflowHeight);
+    const maxY = Math.min(viewport.height, visibleArea.maxY + overflowHeight);
+    const width = maxX - minX;
+    const height = maxY - minY;
+
+    this.#detailArea = { minX, minY, width, height };
+
+    this.reset({ keepCanvas: true });
+  }
+
+  async draw() {
+    // If there is already the lower resolution canvas behind,
+    // we don't show the new one until when it's fully ready.
+    const hideUntilComplete =
+      this.pageView.renderingState === RenderingStates.FINISHED ||
+      this.renderingState === RenderingStates.FINISHED;
+
+    if (this.renderingState !== RenderingStates.INITIAL) {
+      console.error("Must be in new state before drawing");
+      this.reset(); // Ensure that we reset all state to prevent issues.
+    }
+    const { div, pdfPage, viewport } = this.pageView;
+
+    if (!pdfPage) {
+      this.renderingState = RenderingStates.FINISHED;
+      throw new Error("pdfPage is not loaded");
+    }
+
+    this.renderingState = RenderingStates.RUNNING;
+
+    const canvasWrapper = this.pageView._ensureCanvasWrapper();
+
+    const { canvas, prevCanvas, ctx } = this._createCanvas(newCanvas => {
+      // If there is already the background canvas, inject this new canvas
+      // after it. We cannot simply use .append because all canvases must
+      // be before the SVG elements used for drawings.
+      if (canvasWrapper.firstElementChild?.tagName === "CANVAS") {
+        canvasWrapper.firstElementChild.after(newCanvas);
+      } else {
+        canvasWrapper.prepend(newCanvas);
+      }
+    }, hideUntilComplete);
+    canvas.setAttribute("aria-hidden", "true");
+
+    const { width, height } = viewport;
+
+    const area = this.#detailArea;
+
+    const { devicePixelRatio = 1 } = window;
+    const transform = [
+      devicePixelRatio,
+      0,
+      0,
+      devicePixelRatio,
+      -area.minX * devicePixelRatio,
+      -area.minY * devicePixelRatio,
+    ];
+
+    canvas.width = area.width * devicePixelRatio;
+    canvas.height = area.height * devicePixelRatio;
+    canvas.style.position = "absolute";
+    canvas.style.display = "block";
+    canvas.style.width = `${(area.width * 100) / width}%`;
+    canvas.style.height = `${(area.height * 100) / height}%`;
+    canvas.style.top = `${(area.minY * 100) / height}%`;
+    canvas.style.left = `${(area.minX * 100) / width}%`;
+
+    const renderingPromise = this._drawCanvas(
+      {
+        canvasContext: ctx,
+        transform,
+        viewport,
+        pageColors: this.pageColors,
+      },
```

Comment:
Why don't this need to include more of the parameter as in the "full page" rendering, since as-is it *appears* to me that things such as annotationMode, optionalContent, and editingState won't be correctly reflected in the detailView rendering?

https://github.com/mozilla/pdf.js/blob/79a976ce9d49c3a3510588281a386804ebdfb23a/web/pdf_page_view.js#L1070-L1079

Reply:
Do you have an example flow of how I can test the difference?

Reply:
I've just tested this in a PDF document with optionalContent, and it's unfortunately broken just like I imagined so you'll need to add *all* of those parameters to the detailView-`renderContext` as well.

As for testing e.g. the optionalContent stuff:
 - Load any PDF file with such content, e.g. `issue18823.pdf`.
 - Change, e.g. via the sidebar, which optionalContent is visible.
 - Zoom in the document to trigger detailView rendering (to see this easier I manually lowered the `maxCanvasPixels` option first).

Once the detailView renders it uses the initial optionalContent state, rather than the *current* state, which is clearly visible when zooming in/out past the detailView render-threshold.